integrationTests/chainSimulator/staking/stakeAndUnStake_test.go

```diff
@@ -0,0 +1,279 @@
+	err = cm.SetStateMultiple([]*dtos.AddressState{
+		{
+			Address: "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl",
+			Balance: "10000000000000000000000",
+		},
+	})
+	require.Nil(t, err)
+
+	blsKey := "9b7de1b2d2c90b7bea8f6855075c77d6c63b5dada29abb9b87c52cfae9d4112fcac13279e1a07d94672a5e62a83e3716555513014324d5c6bb4261b465f1b8549a7a338bc3ae8edc1e940958f9c2e296bd3c118a4466dec99dda0ceee3eb6a8c"
+
+	// Step 3 --- generate and send a stake transaction with the BLS key of the validator key that was added at step 1
+	stakeValue, _ := big.NewInt(0).SetString("2500000000000000000000", 10)
+	tx := &transaction.Transaction{
+		Nonce:     0,
+		Value:     stakeValue,
+		SndAddr:   newValidatorOwnerBytes,
+		RcvAddr:   rcvAddrBytes,
+		Data:      []byte(fmt.Sprintf("stake@01@%s@010101", blsKey)),
+		GasLimit:  50_000_000,
+		GasPrice:  1000000000,
+		Signature: []byte("dummy"),
+		ChainID:   []byte(configs.ChainID),
+		Version:   1,
+	}
+	stakeTx := helpers.SendTxAndGenerateBlockTilTxIsExecuted(t, cm, tx, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.NotNil(t, stakeTx)
+
+	shardIDValidatorOwner := cm.GetNodeHandler(0).GetShardCoordinator().ComputeId(newValidatorOwnerBytes)
+	accountValidatorOwner, _, err := cm.GetNodeHandler(shardIDValidatorOwner).GetFacadeHandler().GetAccount(newValidatorOwner, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	balanceBeforeActiveValidator := accountValidatorOwner.Balance
+
+	// Step 5 --- create an unStake transaction with the bls key of an initial validator and execute the transaction to make place for the validator that was added at step 3
+	firstValidatorKey, err := cm.GetValidatorPrivateKeys()[0].GeneratePublic().ToByteArray()
+	require.Nil(t, err)
+
+	initialAddressWithValidators := cm.GetInitialWalletKeys().InitialWalletWithStake.Address
+	senderBytes, _ := cm.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(initialAddressWithValidators)
+	shardID := cm.GetNodeHandler(0).GetShardCoordinator().ComputeId(senderBytes)
+	initialAccount, _, err := cm.GetNodeHandler(shardID).GetFacadeHandler().GetAccount(initialAddressWithValidators, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	tx = &transaction.Transaction{
+		Nonce:     initialAccount.Nonce,
+		Value:     big.NewInt(0),
+		SndAddr:   senderBytes,
+		RcvAddr:   rcvAddrBytes,
+		Data:      []byte(fmt.Sprintf("unStake@%s", hex.EncodeToString(firstValidatorKey))),
+		GasLimit:  50_000_000,
+		GasPrice:  1000000000,
+		Signature: []byte("dummy"),
+		ChainID:   []byte(configs.ChainID),
+		Version:   1,
+	}
+	_ = helpers.SendTxAndGenerateBlockTilTxIsExecuted(t, cm, tx, maxNumOfBlockToGenerateWhenExecutingTx)
+
+	// Step 6 --- generate 50 blocks to pass 2 epochs and the validator to generate rewards
+	err = cm.GenerateBlocks(50)
+	require.Nil(t, err)
+
+	validatorStatistics, err := cm.GetNodeHandler(core.MetachainShardId).GetFacadeHandler().ValidatorStatisticsApi()
+	require.Nil(t, err)
+
+	countRatingIncreased := 0
```

Comment:
please extract L143-L152 in a function to be reused

Reply:
done.